autoscale/src/main/java/com/sequenceiq/periscope/service/configuration/LimitsConfigurationService.java

```diff
@@ -0,0 +1,28 @@
+package com.sequenceiq.periscope.service.configuration;
+
+import static com.sequenceiq.periscope.cache.LimitsConfigurationCache.LIMITS_CONFIGURATION_CACHE;
+
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.autoscales.response.LimitsConfigurationResponse;
+import com.sequenceiq.periscope.monitor.evaluator.ScalingConstants;
+import com.sequenceiq.periscope.monitor.handler.CloudbreakCommunicator;
+
+@Service
+public class LimitsConfigurationService {
+
+    @Inject
+    private CloudbreakCommunicator cbCommunicator;
+
+    @Cacheable(cacheNames = LIMITS_CONFIGURATION_CACHE)
+    public Integer getMaxNodeCountLimit() {
+        return Optional.ofNullable(cbCommunicator.getLimitsConfiguration())
+                .map(LimitsConfigurationResponse::getMaxNodeCountLimit)
+                .orElse(ScalingConstants.DEFAULT_CLUSTER_MAX_SIZE);
```

Comment:
if the orElse is ever triggered ... can we please log this?
Also, Given this gets refreshed every 10 minutes - generally log the limit being used. Will make debugging easier.

Reply:
ok